public/javascripts/notebook/observable.js

```diff
@@ -113,31 +113,48 @@ return new function () {
 
 
 
-  this.makeObservableHelper = function (id, kind, initialValue) {
+  this.makeObservableHelper = function (id, kind, initialValue, df) {
+    var _df = df
+    if (_.isUndefined(_df)) {
+      _df = function(publishValue) {
+        this.df.done(function(ob) {
+          ob(publishValue);
+        });
+      }
+      _df.df = $.Deferred();
+      _df.subscribe = function(subscriber) {
+        this.df.done(function(ob) {
+          ob.subscribe(subscriber);
+        });
+      };
+    }
     if(!this.isInitialized()) {
       // in case the Observable isn't initialized, we differ the creation of the observable
       var args = Array.prototype.slice.call(arguments);
       var me = this;
+      args.push(_df);
```

Comment:
this is a bit fishy.
1. we pass less arguments than required when calling this fn: `return this.makeObservableHelper(id, 'observable', initialValue)`
2. not quite sure what happens in `Array.prototype.slice.call(arguments);`, but doing only `args.push(_df)` isn't that we always assume to receive less arguments than required ?